openproblems/tasks/cell_cell_communication/metrics/odds_ratio.py

```diff
@@ -0,0 +1,29 @@
+from ....tools.decorators import metric
+
+import numpy as np
+import scipy.stats as stats
+
+
+@metric(metric_name="Odds Ratio", maximize=True)
+def odds_ratio(adata, top_n=100):
+    # Join benchmark (assumed truth) and ccc results
+    # Get /w ccc_results and a response [0, 1] column
+    gt = adata.uns["bench"].merge(adata.uns["ccc"], how="right")
+    gt = gt[gt["response"].notna()]
+
+    # assign the top rank interactions to 1
+    a = np.zeros(len(gt["score"]))
+    a[0:top_n] = 1
+    gt.loc[:, ["top_n"]] = a
+
+    # Shape to contingency table
+    table = np.array(gt.pivot_table(index=["top_n", "response"], aggfunc="size"))
+
+    # if positive or negative class is not in top_n
+    if table.shape != (4,):
```

Comment:
Looks like the unit tests never make it past this stage. Not sure this is necessarily a problem but worth flagging

Reply:
Any idea why this might be the case?

Reply:
Because we're working with toy data it's unsurprising that the results are "bad" -- which I guess means it's more likely that you don't get any hits in the top n. It's a little concerning from a test coverage perspective though. Maybe you can write a specific test (see the test_dimensionality_reduction.py for a task-specific test example) to check that it generates the right output when handed a "good" test case? (i.e. hand craft some good results with an expected outcome)

Reply:
Hmmm... the 'bad' case should be covered by `if table.shape != (4,):`. If either of TP, FP, TN, and FN values for the odds ratio (two classes) is not present then table.shape is not what is expected and the function simply returns a 1 (=random).

Reply:
All tests seem to pass now on the CI, previously test_3 if I remember correctly was failing. Perhaps, this is resolved - likely related to specifically declaring response as int?

Reply:
Unfortunately still not covered (see [codecov](https://app.codecov.io/gh/openproblems-bio/openproblems/compare/520/diff#D10L27)). The "bad" case is indeed covered -- it's the good case that never gets tested. I'd suggest creating a test in a new test file `test_4_cell_cell_comm.py` which basically does the following:
```
adata = cell_cell_comm.api.sample_dataset()
adata.uns['ccc_pred'] = adata.uns['ccc_target']
odds_ratio = cell_cell_comm.metrics.odd_ratio(adata)
assert odds_ratio == 1
```
or something like that. You can copy `test_4_dimensionality_reduction` for a template.

Reply:
OK. I found where the issue was and fixed it! Now it should test the 'good' case. :)